torch/nn/modules/module.py

```diff
@@ -391,6 +408,8 @@ def remove_from(*dicts):
                     raise AttributeError(
                         "cannot assign module before Module.__init__() call")
                 remove_from(self.__dict__, self._parameters, self._buffers)
+                if value is not None:
+                    value._name = name
```

Comment:
Treating modules like this is a bit awkward, because we don't enforce the model to form a tree - it's more of a DAG. The same module can be a submodule of multiple parents, and it happens in a few models. A better solution would be to keep a stack of modules, and inside that `slow_forward()` function I mentioned you could look up the topmost module, fine the name of the current module in the outer scope, and use that. This would let modules have multiple names and remove the need for this attribute. And it's tracing so we don't care this much about perf

Reply:
Done